src/main/java/com/xiaomi/infra/pegasus/client/ClientOptions.java

```diff
@@ -379,6 +401,28 @@ public Builder serviceFQDN(String serviceFQDN) {
       return this;
     }
 
+    /**
+     * kerberos keytab. Defaults to {@literal ""}, see {@link #DEFAULT_KEYTAB}.
+     *
+     * @param keyTab
+     * @return {@code this}
+     */
+    public Builder keyTab(String keyTab) {
+      this.keyTab = keyTab;
+      return this;
+    }
+
+    /**
+     * kerberos principal. Defaults to {@literal ""}, see {@link #DEFAULT_PRINCIPAL}.
+     *
+     * @param principal
+     * @return {@code this}
+     */
+    public Builder principal(String principal) {
+      this.principal = principal;
+      return this;
+    }
```

Comment:
I'd recommend you wrap these two options into a Credential class in order to support more types of authentication with less modification. Your current implementation limits the auth method to Kerberos only. The SecurityReplicaSessionInterceptor also assumes that the mechanism is definitely Kerberos. You should take extensibility into account.

Consider this design. A Credential could possibly be keytab/principle, or AK/SK that's provided by the cloud provider. You can extend Credential to a KerberosCredential for Kerberos authentication.

```java
public interface Credential {
  public Mechanism getMechanism();
}

public interface Mechanism {
  void authenticate() throws PException;
}
```

To load from the configuration file, KerberosCredential must have a constructor that accepts Configuration:

```java
class KerberosCredential {
  KerberosCredential(Configuration cfg);

  Mechanism getMechanism() {
    return new JAASMechanism(this);
  }
}

class JAASMechanism {
    void authenticate() throws PException; // use LoginContext
}
```

To determine the auth method we use, an option "auth_mechanism=kerberos" must be given. It can replace "enable_auth" for skipping auth. "auth_mechanism=none" or "auth_mechanism=" mean no auth.